service-base/src/test/java/org/eclipse/hono/service/amqp/AmqpServiceBaseTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ *
+ */
+package org.eclipse.hono.service.amqp;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonReceiver;
+import org.apache.qpid.proton.amqp.transport.Target;
+import org.apache.qpid.proton.engine.Record;
+import org.apache.qpid.proton.engine.impl.RecordImpl;
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.auth.Activity;
+import org.eclipse.hono.auth.HonoUser;
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.service.auth.AuthorizationService;
+import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.util.ResourceIdentifier;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests verifying behavior of {@link AmqpServiceBase}.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class AmqpServiceBaseTest {
+    private static final String CON_ID = "connection-id";
+    private static final String ENDPOINT = "anEndpoint";
+
+    private Vertx vertx;
+    private EventBus eventBus;
+
+    /**
+     * Sets up common mock objects used by the test cases.
+     */
+    @Before
+    public void initMocks() {
+        eventBus = mock(EventBus.class);
+        vertx = mock(Vertx.class);
+        when(vertx.eventBus()).thenReturn(eventBus);
+    }
+
+    private AmqpServiceBase<ServiceConfigProperties> createServer(final Endpoint amqpEndpoint) {
+
+        AmqpServiceBase<ServiceConfigProperties> server = new AmqpServiceBase<ServiceConfigProperties>() {
+            @Override
+            protected void onRemoteConnectionOpen(ProtonConnection connection) {
+            }
+
+            @Override
+            protected void onRemoteConnectionOpenInsecurePort(ProtonConnection connection) {
+            }
+        };
+        server.setConfig(new ServiceConfigProperties());
+        if (amqpEndpoint != null) {
+            server.addEndpoint(amqpEndpoint);
+        }
+        server.init(vertx, mock(Context.class));
+        return server;
+    }
+
+    @Test
+    public void testHandleReceiverOpenForwardsToTelemetryEndpoint() throws InterruptedException {
```

Comment:
we should remove `Telemetry` from this method's name since it doesn't have anything to do with Telemetry anymore